pyaedt/generic/ibis_reader.py

```diff
@@ -0,0 +1,748 @@
+
+    @property
+    def clamp(self):
+        """Clamp."""
+        return self._clamp
+
+    @clamp.setter
+    def clamp(self, value):
+        self._clamp = value
+
+    @property
+    def enable(self):
+        """Is model enabled or not."""
+        return self._enable
+
+    @enable.setter
+    def enable(self, value):
+        self._enable = value
+
+
+class Ibis:
+    """Ibis model with all data extracted: name, components, models.
+
+    Parameters
+    ----------
+    name : str
+        Name of ibis model.
+    circuit : class:`pyaedt.circuit.Circuit`
+        Circuit in which the ibis components will be used.
+    """
+
+    # Ibis reader must work independently or in Circuit.
+    def __init__(self, name, circuit):
+        self.circuit = circuit
+        self._name = name
+        self._components = {}
+        self._model_selectors = []
+        self._models = []
+
+    @property
+    def name(self):
+        """Name of the ibis model."""
+        return self._name
+
+    @property
+    def components(self):
+        """List of all components included in the ibis file."""
+        return self._components
+
+    @components.setter
+    def components(self, value):
+        self._components = value
+
+    @property
+    def model_selectors(self):
+        """List of all model selectors included in the ibis file."""
+        return self._model_selectors
+
+    @model_selectors.setter
+    def model_selectors(self, value):
+        self._model_selectors = value
+
+    @property
+    def models(self):
+        """List of all models included in the ibis file."""
+        return self._models
+
+    @models.setter
+    def models(self, value):
+        self._models = value
+
+    @property
+    def buffers(self):
+        """Buffers included into the ibis model."""
+        return self._buffers
+
+    @buffers.setter
+    def buffers(self, value):
+        self._buffers = value
+
+
+class IbisReader:
+    def read_project(self, filename, circuit):
+        """Reads *.ibis file content.
+
+        Parameters
+        ----------
+        filename : str
+            Name of ibis model.
+        circuit : class:`pyaedt.circuit.Circuit`
+            Circuit in which the ibis components will be used.
+
+        Returns
+        ----------
+        :class:`pyaedt.generic.ibis_reader.Ibis`
+            Ibis object exposing all data from the ibis file.
+        """
+
+        if os.path.exists(filename) == False:
+            error_message = filename + " does not exist."
+            raise FileExistsError(error_message)
+
+        ibis_name = pyaedt.generic.general_methods.get_filename_without_extension(filename)
+        ibis = Ibis(ibis_name, circuit)
+
+        # Read *.ibis file.
+        with open(filename, "r") as file:
+            while True:
+                current_line = file.readline()
+                if not current_line:
+                    break
+
+                if self.IsStartedWith(current_line, "[Component] ") == True:
```

Comment:
There’s no reason to compare with `True`, the `if` statement will still work. `if True == True:` is the same as `if True:`, though even if you were comparing a Boolean with a Boolean, you would use `is` rather than `==`.
```suggestion
                if self.IsStartedWith(current_line, "[Component] "):
```

Reply:
Make sense.

Reply:
I fixed this issue in the entire file.